crates/rspack_core/src/loader/loader_runner.rs

```diff
@@ -17,7 +18,11 @@ pub struct CompilerContext {
   pub module_context: Option<Box<Context>>, // current module context
   pub module_source_map_kind: SourceMapKind,
 
-  pub queue_handler: Option<QueueHandler>,
+  pub factorize_queue: Option<FactorizeQueueHandler>,
```

Comment:
Maybe init the `XXXQueue` here instead of the handler  when a new compilation is created? This could help us remove `Option`.

Reply:
Because handler is `tx`, if we init them during compilation creation, we have to put `rx` into compilation as well, and some API like `handle_module_creation` needs &Compilation and &mut FactorizeQueue, If I change the API from taking &Compilation and &mut FactorizeQueue to only taking &mut Compilation, some other upper stream API needs to change as well because them don't have &mut Compilation :(.
I encountered this a lot before, so I choose the easier way here